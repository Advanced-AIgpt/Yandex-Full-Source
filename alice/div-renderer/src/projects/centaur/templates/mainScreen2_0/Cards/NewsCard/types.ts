import { IAbstractCardProps } from '../types';

export interface INewsCardProps extends IAbstractCardProps {
    type: 'news';
    title?: string | null;
    content?: string | null;
    image?: string | null;
}
