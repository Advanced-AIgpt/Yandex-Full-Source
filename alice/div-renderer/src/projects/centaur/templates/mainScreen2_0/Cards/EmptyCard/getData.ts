import { IEmptyCardProps } from './types';
import { ICardDataAdapter } from '../types';

export const getEmptyCardData: ICardDataAdapter<IEmptyCardProps> = function getEmptyCardData(card, requestState) {
    if (card && card.VacantCardData) {
        return {
            type: 'empty',
            requestState,
        };
    }
    return null;
};
