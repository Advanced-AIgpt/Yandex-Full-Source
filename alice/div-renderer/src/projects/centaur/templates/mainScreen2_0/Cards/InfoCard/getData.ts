import { sample } from 'lodash';
import { IInfoCardProps } from './types';
import { DISCOVERY_CARDS } from '../../../mainScreen/discovery';
import { ICardDataAdapter } from '../types';
import { textAction } from '../../../../../../common/actions';

export const getInfoCardData: ICardDataAdapter<IInfoCardProps> = function getInfoCardData(card, requestState) {
    if (typeof card.InfoCardData !== 'undefined' && card.InfoCardData !== null) {
        const data = card.InfoCardData.Title === 'test' ? DISCOVERY_CARDS[0] : sample(DISCOVERY_CARDS) || DISCOVERY_CARDS[0];

        return {
            type: 'info',
            title: data.title,
            image_background: data.bg,
            description: data.description,
            requestState,
            actions: [
                {
                    log_id: 'discovery.mainscreen.execute',
                    url: textAction(data.description),
                },
            ],
        };
    }
    return null;
};
