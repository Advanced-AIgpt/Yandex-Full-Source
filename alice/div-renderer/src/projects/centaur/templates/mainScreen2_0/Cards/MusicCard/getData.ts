import { IMusicCardProps } from './types';
import { colorDarkGrey } from '../../../../style/constants';
import { ICardDataAdapter } from '../types';

const playlists: readonly [string, string][] = [
    ['Плейлист дня', '#31AD54'],
    ['Мне нравится', '#E53866'],
    ['Тайник', '#186FD3'],
    ['Подкасты', '#036292'],
    ['Дежавю', '#A019D1'],
    ['Премьера', '#D1673E'],
    ['Плейлист с Алисой', '#2D0B87'],
    ['Мой 2021', '#C20146'],
    ['Плейлист семьи', '#FE8853'],
];

export const getMusicCardData: ICardDataAdapter<IMusicCardProps> = function getMusicCardData(card, requestState) {
    if (typeof card.MusicCardData !== 'undefined' && card.MusicCardData !== null) {
        const data = card.MusicCardData;

        const knownPlaylist = playlists.find(el => el[0] === data.Name);

        const color = knownPlaylist ? knownPlaylist[1] : colorDarkGrey;

        return {
            type: 'music',
            name: data.Name,
            modified: data.Modified?.value,
            color,
            cover: data.Cover,
            actions: (card.Action && [
                {
                    log_id: 'music_card_action',
                    url: card.Action,
                },
            ]) || undefined,
            preventDefaultLoader: true,
            requestState,
        };
    }
    return null;
};
