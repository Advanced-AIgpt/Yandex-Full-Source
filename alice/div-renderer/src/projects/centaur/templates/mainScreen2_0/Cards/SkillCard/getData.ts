import { ISkillCardProps } from './types';
import { ICardDataAdapter } from '../types';
import { NAlice } from '../../../../../../protos';
import { EnumLayer } from '../../../../actions/client';
type ITButton = NAlice.NData.TCentaurWidgetCardData.TExternalSkillCardData.TMainScreenData.ITButton;

interface INormalisedButton {
    Text: string;
    Payload?: string;
}

function isNormalButton(el: ITButton): el is INormalisedButton {
    return Boolean(el.Text);
}

export const getSkillCardData: ICardDataAdapter<ISkillCardProps> = function getSkillCardData(card, requestState) {
    if (typeof card.ExternalSkillCardData !== 'undefined' && card.ExternalSkillCardData !== null) {
        const data = card.ExternalSkillCardData;

        if (data.widgetGalleryData) {
            return {
                type: 'skill',
                requestState,
                skillName: data.skillInfo?.Name || ' ',
                skillImage: data.skillInfo?.Logo || ' ',
                skillId: data.skillInfo?.SkillId || ' ',
                title: ' ',
            };
        } else {
            return {
                type: 'skill',
                requestState,
                title: data.mainScreenData?.Title,
                comment: data.mainScreenData?.Text,
                image: data.mainScreenData?.ImageUrl,
                skillName: data.skillInfo?.Name || ' ',
                skillImage: data.skillInfo?.Logo || ' ',
                skillId: data.skillInfo?.SkillId || ' ',
                layer: EnumLayer.dialog,
                buttons: data.mainScreenData?.buttons
                    ?.filter(isNormalButton)
                    .map(el => ({
                        text: el.Text,
                        payload: el.Payload || undefined,
                    })),
            };
        }
    }
    return null;
};
