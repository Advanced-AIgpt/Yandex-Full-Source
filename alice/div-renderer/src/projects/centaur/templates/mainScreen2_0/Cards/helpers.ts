export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
