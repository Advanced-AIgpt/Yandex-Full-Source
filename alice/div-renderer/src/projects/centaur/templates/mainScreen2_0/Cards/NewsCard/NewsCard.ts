import { Div, ImageBackground, TextBlock } from 'divcard2';
import { text28m } from '../../../../style/Text/Text';
import { backgroundToImageWithText, colorWhiteOpacity90 } from '../../../../style/constants';
import { INewsCardProps } from './types';
import { BasicTextCard } from '../BasicTextCard';
import { logger } from '../../../../../../common/logger';
import BasicErrorCard from '../BasicErrorCard';
import { Avatar } from '../../../../../../common/helpers/avatar';
import { basicCardBackground } from '../BasicCard';
import { getCardMainScreenId } from '../helpers';

export default function NewsCard({
    title,
    content,
    image,
    actions,
    longtap_actions,
    colIndex,
    rowIndex,
}: INewsCardProps): Div {
    if (!title || !content) {
        logger.error(new Error('The news card should have a title and description, but some of this was not transmitted.'));

        return BasicErrorCard({
            colIndex,
            rowIndex,
            title: 'Новости',
            description: 'Ошибка сети. Не могу показать новости. Проверьте интернет.',
            actions,
            longtap_actions,
        });
    }

    const cardData: Parameters<typeof BasicTextCard>[0] = {
        id: getCardMainScreenId({ rowIndex, colIndex }),
        title,
        items: [
            new TextBlock({
                ...text28m,
                max_lines: 5,
                text_color: colorWhiteOpacity90,
                text: content,
            }),
        ],
        actions,
        longtap_actions,
    };

    if (image) {
        const avatar = Avatar.fromUrl(image);

        cardData.background = [
            ...basicCardBackground,
            new ImageBackground({
                image_url: avatar && avatar.namespace === 'ynews' ? avatar.setTypeName('380x214').toString() : image,
                scale: 'fill',
                preload_required: 1,
                content_alignment_horizontal: 'center',
                content_alignment_vertical: 'center',
            }),
            ...backgroundToImageWithText,
        ];
    }

    return BasicTextCard(cardData);
}
