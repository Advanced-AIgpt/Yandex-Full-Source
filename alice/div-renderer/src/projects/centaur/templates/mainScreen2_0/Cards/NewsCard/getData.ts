import { compact } from 'lodash';
import { INewsCardProps } from './types';
import { EnumLayer } from '../../../../actions/client';
import { getImageByTopic } from '../../../../helpers/newsHelper';
import { ICardDataAdapter } from '../types';

export const getNewsCardData: ICardDataAdapter<INewsCardProps> = function getNewsCardData(card, requestState) {
    if (typeof card.NewsCardData !== 'undefined' && card.NewsCardData !== null) {
        const data = card.NewsCardData;

        return {
            type: 'news',
            title: data.Title,
            content: data.Content,
            image: data.ImageUrl || getImageByTopic(data.Topic) || undefined,
            layer: EnumLayer.dialog,
            requestState,
            actions: compact([
                card.Action && {
                    log_id: 'news_card_action',
                    url: card.Action,
                },
            ]),
        };
    }
    return null;
};
