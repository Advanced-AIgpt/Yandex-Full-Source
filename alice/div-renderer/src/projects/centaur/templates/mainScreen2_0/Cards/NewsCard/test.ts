import NewsCard from './NewsCard';
import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';
import { NAlice } from '../../../../../../protos';
import { getNewsCardData } from './getData';
import { logger } from '../../../../../../common/logger';
const TCardData = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn.TCardData;

jest.mock('../../../../../../common/logger');

describe('Main screen empty card', () => {
    it('should get actual data', () => {
        expect(getNewsCardData(new TCardData({}))).toBeNull();

        expect(getNewsCardData(new TCardData({
            YouTubeCardData: {},
        }))).toBeNull();

        expect(getNewsCardData(new TCardData({
            YouTubeCardData: {},
            NewsCardData: {
            },
        }))).not.toBeNull();

        expect(getNewsCardData(new TCardData({
            NewsCardData: {
                Title: 'Медуза',
                Content: 'Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger',
            },
        }))).not.toBeNull();
    });

    it('should match snapshot', () => {
        expect(AnonymizeDataForSnapshot(NewsCard({
            type: 'news',
            title: 'Медуза',
            content: 'Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalledTimes(0);
    });
    it('should match snapshot with error', () => {
        expect(AnonymizeDataForSnapshot(NewsCard({
            type: 'news',
            title: '',
            content: 'Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalled();
    });
});
