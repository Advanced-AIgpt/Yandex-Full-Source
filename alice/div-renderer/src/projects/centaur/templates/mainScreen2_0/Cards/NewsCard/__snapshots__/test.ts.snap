// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen empty card should match snapshot 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "font_size": 32,
          "font_weight": "medium",
          "line_height": 40,
          "text": "Медуза",
          "text_color": "#80ffffff",
          "type": "text",
        },
      ],
      "orientation": "horizontal",
      "type": "container",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "font_size": 28,
      "font_weight": "medium",
      "line_height": 36,
      "max_lines": 5,
      "text": "Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger",
      "text_color": "#e6ffffff",
      "type": "text",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;

exports[`Main screen empty card should match snapshot with error 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "font_size": 32,
          "font_weight": "medium",
          "line_height": 40,
          "text": "Новости",
          "text_color": "#80ffffff",
          "type": "text",
        },
      ],
      "orientation": "horizontal",
      "type": "container",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "font_size": 28,
      "font_weight": "medium",
      "line_height": 36,
      "text": "Ошибка сети. Не могу показать новости. Проверьте интернет.",
      "text_color": "#80ffffff",
      "type": "text",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;
